DynatraceAppMon/Context.cs

```diff
@@ -330,6 +430,38 @@ public ArrayList getMethod(Lookup lookup, List<Project> projects)
             return codeFunctions;
         }
 
+        private void FindClasses(Lookup lookup, ProjectItems projectItems, ref List<String> hits)
+        {
+            foreach (ProjectItem item in projectItems)
+            {
+                try
+                {
+                    if (item.ProjectItems != null) FindClasses(lookup, item.ProjectItems, ref hits);
+                    else if (!HasClassFiles(item)) continue;
+                    for (short i = 1; i<=item.FileCount; i++)
+                    {
+                        SyntaxTree tree = CSharpSyntaxTree.ParseText(File.ReadAllText(item.FileNames[i]));
+                        var root = (CompilationUnitSyntax)tree.GetRoot();
+                        //get all classes from file
+                        var classes = root.DescendantNodes()
+                            .OfType<ClassDeclarationSyntax>()
+                            .ToList();
+                        foreach (var clazz in classes)
+                        {
+                            if (lookup.typeName.Contains(clazz.Identifier.ToString()))
+                            {
+                                hits.AddRange(GetClassFiles(item));
+                            }
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    log(LOG_ERROR + e.ToString());
```

Comment:
what kind of exception may occur ? can we safely proceed thereafter ?

Reply:
It's not easy to predict excpetions thrown from COM objects. That's the exception catching schema accepted throughout the plugin.